lib/overcommit/hook/pre_commit/metadata_json_lint.rb

```diff
@@ -0,0 +1,50 @@
+module Overcommit::Hook::PreCommit
+  #
+  # Run's the Puppet metadata linter. It has support for adding options
+  # in the .overcommit.yaml
+  #
+  # PreCommit:
+  #   MetadataJsonLint:
+  #     enabled: true
+  #     strict_license: false
+  #     strict_dependencies: false
+  #     fail_on_warning: true
+  #     description: 'Checking module metadata'
+  #
+  # @see https://voxpupuli.org/blog/2014/11/06/linting-metadata-json/
+  #
+  class MetadataJsonLint < Base
+    MESSAGE_REGEX = /\((?<type>.*)\).*/
+
+    MESSAGE_TYPE_CATEGORIZER = lambda do |type|
+      type == 'WARN' ? :warning : :error
+    end
+
+    def options
+      [:strict_license, :strict_dependencies, :fail_on_warning].collect do |option|
+        name = option.to_s.tr('_', '-')
+        value = config.fetch(option.to_s) { true }
+        if value
+          "--#{name}"
+        else
+          "--no-#{name}"
+        end
+      end
+    end
```

Comment:
I would recommend against making these configurable options on the hook itself since flag names can change and you can simply add/remove flags using the `flags` option for the hook.

So for your configuration, I would change the YAML to say:

```yaml
  PuppetMetadataJsonLint:
    enabled: false
    description: 'Checking module metadata'
    flags: ['--no-strict_license', '--no-strict-dependencies', '--fail-on-warning']
    include: 'metadata.json'
```

This reduces the surface area you need to support, so if the underlying tool changes you don't need to make any changes to Overcommit to start using them.

This has the added benefit of simplifying your implementation to not require this `options` method at all:

```ruby
def run 
  result = execute(command, args: applicable_files)
  output = result.stdout.chomp.gsub(/^"|"$/, '')
  return :pass if result.success? && output.empty?

  extract_messages(
    output.split("\n"),
    MESSAGE_REGEX,
    MESSAGE_TYPE_CATEGORIZER
  )
end
```

Since you specified `include` in the configuration, `applicable_files` will always just include the one `metadata.json` file.